### PR TITLE
Reduce binary size of WebKit Linux build bundles (GTK and WPE)

### PR DESCRIPTION
**Pull request by clopez**
This patch changes the build system to use the JHBuild minimal dependency
system introduced in WebKit r264092 <https://trac.webkit.org/r264092>

The build has been tested with Ubuntu-18.04

The binary size of the zip bundles is now: 37M (WPE) and 40M (GTK).
Previously it was 54M and 59M (respectively)

### CODE REVIEW

File: browser_patches/webkit/archive.sh [hunk 1]
```diff
@@ -75,11 +75,15 @@ createZipForLinux() {
     # Copy libvpx.so.5 as Ubuntu 20.04 comes with libvpx.so.6
     ldd WebKitBuild/WPE/Release/bin/MiniBrowser | grep -o '[^ ]*\/libvpx.so.5[^ ]*' | xargs cp -t $tmpdir
     # Copy some wayland libraries required for Web Process t
-    cp -d -t $tmpdir WebKitBuild/WPE/DependenciesWPE/Root/lib/libva\-*
+    if ls WebKitBuild/WPE/DependenciesWPE/Root/lib/libva\-* 2>&1 >/dev/null; then
```
Comment (aslushnikov) [inline browser_patches/webkit/archive.sh:78]: why is this condition added? Does it happen now that there's nothing to copy?

File: browser_patches/webkit/archive.sh [hunk 2]
```diff
@@ -75,11 +75,15 @@ createZipForLinux() {
...
+      cp -d -t $tmpdir WebKitBuild/WPE/DependenciesWPE/Root/lib/libva\-*
+    fi
     # Injected bundle is loaded dynamicly via dlopen => not bt listed by ldd.
     cp -t $tmpdir WebKitBuild/WPE/Release/lib/libWPEInjectedBundle.so
-    mkdir -p $tmpdir/gio/modules
-    cp -t $tmpdir/gio/modules $PWD/WebKitBuild/WPE/DependenciesWPE/Root/lib/gio/modules/*
+    if test -d $PWD/WebKitBuild/WPE/DependenciesWPE/Root/lib/gio/modules/; then
```
Comment (aslushnikov) [inline browser_patches/webkit/archive.sh:83]: I assume that this if-condition means that there are no gio modules to copy if we build with Ubuntu-18.

However, I remember not having these were breaking our "ignore https errors" tests on Github Actions.

Anyhow, we can land this and then see how it works on our CI!

**Review by aslushnikov — APPROVED** [review]

File: browser_patches/webkit/archive.sh (hunk 1 above)
Comment (clopez) [reply browser_patches/webkit/archive.sh:78]: I tried to adapt the script so it works both with this minimal jhbuild moduleset and with the default one (which builds many other libraries). When using this minimal moduleset, the if condition will be false.

File: browser_patches/webkit/archive.sh (hunk 2 above)
Comment (clopez) [reply browser_patches/webkit/archive.sh:83]: That's right, with the minimal jhbuild moduleset no extra gio modules are built, only the system ones are used. The script however is adapted to check for any gio module built and copy it (if its there)

**Comment by aslushnikov** [comment]
@clopez awesome. Can you please re-baseline the PR? We'll land it and try rolling the fresh builds in to see how it works on our CI.

**Comment by clopez** [comment]
> @clopez awesome. Can you please re-baseline the PR? We'll land it and try rolling the fresh builds in to see how it works on our CI.

@aslushnikov: done.. i have rebased, and force pushed it.

File: browser_patches/webkit/UPSTREAM_CONFIG.sh [hunk 3]
```diff
@@ -1,3 +1,3 @@
 REMOTE_URL="https://github.com/webkit/webkit"
 BASE_BRANCH="master"
-BASE_REVISION="19d11039faa6e4da8cf7487e9c54a00a7c6e18e2"
+BASE_REVISION="822da5711ad337e1a7da82e813aa99e14fb79e49"
```
Comment (aslushnikov) [inline browser_patches/webkit/UPSTREAM_CONFIG.sh:3]: Is this an intended change? Yury rolled webkit recently, so there should be no need.

Comment (clopez) [reply browser_patches/webkit/UPSTREAM_CONFIG.sh:3]: Is intended. `19d11039faa6e4da8cf7487e9c54a00a7c6e18e2` is webkit **r264022**, but for this patch we need at least webkit **r264092** (which its like 70 commits ahead). Here I used a commit a bit later than that: `822da5711ad337e1a7da82e813aa99e14fb79e49` (**r264108**) which was the last commit i could assert yesterday it was passing fine at the build.webkit.org CI

**Pull request merged**